engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -383,72 +383,42 @@ - (void)clipRRect:(const flutter::DlRoundRect&)clipDlRRect matrix:(const flutter
 - (void)clipPath:(const flutter::DlPath&)dlPath matrix:(const flutter::DlMatrix&)matrix {
   containsNonRectPath_ = YES;
   CGMutablePathRef pathRef = CGPathCreateMutable();
-  bool subpath_needs_close = false;
-  std::optional<flutter::DlPoint> pending_moveto;
 
-  auto resolve_moveto = [&pending_moveto, &pathRef]() {
-    if (pending_moveto.has_value()) {
-      CGPathMoveToPoint(pathRef, nil, pending_moveto->x, pending_moveto->y);
-      pending_moveto.reset();
-    }
+  flutter::DlPathReceiver receiver{
+      .path_info =
+          [](flutter::DlPathFillType type, bool is_convex) {
+            // CGPaths do not have an inherit fill type, we would need to remember
+            // the fill type and employ it when we use the path.
```

Comment:
A missing fill type bug here with an even bigger mystery as I don't think there's a way to attach a fill type to a path in CGPath - it seems to be a graphics environment setting...?

Reply:
Worth filling a bug for. I suspect that even odd clipping is probably not used

Reply:
Perhaps we'd just need a stack of fill types to maintain in parallel with the stack of CGPath objects?

Reply:
https://github.com/flutter/flutter/issues/164826

Reply:
link to issue added in latest commit.